apis/v1alpha1/clusterprofile_types.go

```diff
@@ -0,0 +1,125 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ClusterProfileSpec defines the desired state of ClusterProfile.
+type ClusterProfileSpec struct {
+	// DisplayName defines a human-readable name of the ClusterProfile
+	// +optional
+	DisplayName string `json:"displayName"`
+
+	// ClusterManager defines which cluster manager owns this ClusterProfile resource
+	// +required
+	ClusterManager ClusterManager `json:"clusterManager"`
+}
+
+// ClusterManager defines which cluster manager owns this ClusterProfile resource.
+// This field is immutable.
+// It's recommended that each cluster manager instance should set a different values to this field.
+// In addition, it's recommended that a predefined label with key "x-k8s.io/cluster-manager"
+// should be added by the cluster manager upon creation. See constant LabelClusterManager.
+// The value of the label should be the same as the name of the cluster manager.
+// The purpose of this label is to make filter clusters from different cluster managers easier.
+//
+// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="ClusterManager is immutable"
+type ClusterManager struct {
+	// Name defines the name of the cluster manager
+	// +required
+	Name string `json:"name"`
+}
+
+// ClusterProfileStatus defines the observed state of ClusterProfile.
+type ClusterProfileStatus struct {
+	// Conditions contains the different condition statuses for this cluster.
+	Conditions []metav1.Condition `json:"conditions"`
+
+	// Version defines the version information of the cluster.
+	Version ClusterVersion `json:"version,omitempty"`
+
+	// Properties defines properties of collected from the cluster.
+	// For example, a unique cluster identifier (id.k8s.io).
+	// The set of properties is not uniform across a fleet, some properties can be
+	// vendor or version specific and may not be included from all clusters.
+	// +optional
+	Properties []Property `json:"properties,omitempty"`
+}
+
+// ClusterVersion represents version information about the cluster.
+type ClusterVersion struct {
+	// Kubernetes is the kubernetes version of the cluster.
+	// +optional
+	Kubernetes string `json:"kubernetes,omitempty"`
+}
+
+// Property defines a property collected from a cluster.
+type Property struct {
+	// Name is the name of a property resource on cluster. It's a well-known
+	// or customized name to identify the propertie.
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:validation:MinLength=1
+	Name string `json:"name"`
+
+	// Value is a property-dependent string
+	// +kubebuilder:validation:MaxLength=1024
+	// +kubebuilder:validation:MinLength=1
+	Value string `json:"value"`
+}
+
+const (
+	// ClusterConditionJoined means the cluster has successfully joined the control.
```

Comment:
It isn't clear what "Joined" really conveys. Doesn't the ClusterProfile CR's existence means that it has "Joined" the list of ClusterProfiles?

Reply:
Yes, this is a confusing condition. Originally, we wanted a condition to highlight the fact that a cluster is now "registered" into a cluster manager. I guess we are not ready to flush out all that details yet so I removed this constant for now.

Reply:
No strong feelings on if this is actually needed, but if we do want to include a status like this `Registered` (or `Accepted` which is commonly used for resources over in Gateway API) feel like better options than `Joined`.

Reply:
I've removed this condition for now . If we want to bring it back as Registered or Accepted (prefer accepted because it's used in gateway API), we can reopen this thread.